bundle/regal/rules/bugs/inconsistent_args.rego

```diff
@@ -0,0 +1,63 @@
+# METADATA
+# description: Inconsistently named function arguments
+package regal.rules.bugs["inconsistent-args"]
+
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+import data.regal.ast
+import data.regal.result
+
+report contains violation if {
+	count(ast.functions) > 0
+
+	# Comprehension indexing — as made obvious here it would be great
+	# to have block level scoped ignore directives...
+	function_args_by_name := {name: args_list |
+		some i
+
+		# regal ignore:prefer-some-in-iteration
+		name := ast.ref_to_string(ast.functions[i].head.ref)
+		args_list := [args |
+			some j
```

Comment:
Probably won't make a dent on performance, but we could add a `i != j` here, right?